src/go/rdctl/cmd/api.go

```diff
@@ -0,0 +1,199 @@
+		}
+	}
+	for _, x := range apiSettings.RawFields {
+		if strings.HasPrefix(x, "body=") {
+			if len(apiSettings.CookedFields) > 0 || len(apiSettings.RawFields) > 1 {
+				return "", fmt.Errorf("%s", "when a body field is specified, no other fields may be specified")
+			}
+			return strings.SplitN(x, "=", 2)[1], nil
+		}
+	}
+	return "", nil
+}
+
+func getBody() (*bytes.Buffer, error) {
+	bodyTemplate, err := getNamedBodyTemplate()
+	if err != nil {
+		return nil, err
+	}
+	if bodyTemplate != "" {
+		return bytes.NewBufferString(bodyTemplate), nil
+	}
+	if len(apiSettings.RawFields) == 0 && len(apiSettings.CookedFields) == 0 {
+		return nil, nil
+	}
+
+	containsNonDigit := regexp.MustCompile(`\D`)
+	for _, s := range apiSettings.CookedFields {
+		parts := strings.SplitN(s, "=", 2)
+		if parts[1] == "true" || parts[1] == "false" || containsNonDigit.FindString(parts[1]) == "" {
+			apiSettings.RawFields = append(apiSettings.RawFields, s)
+		} else if parts[1][0] == '"' && strings.HasSuffix(parts[1], `"`) {
+			apiSettings.RawFields = append(apiSettings.RawFields, s)
+		} else {
+			apiSettings.RawFields = append(apiSettings.RawFields, fmt.Sprintf(`%s="%s"`, parts[0], parts[1]))
+		}
+	}
+
+	bufTemplate := new(bytes.Buffer)
+	bufTemplate.WriteString(`{ "kubernetes": { `)
+	values := make(map[string]interface{})
+	commaOrEmptyString := ""
+
+	for _, s := range apiSettings.RawFields {
+		parts := strings.SplitN(s, "=", 2)
+		canonicalName := ""
+		switch(parts[0]) {
+		case "container-engine":
+			canonicalName = "engine"
+		case "kubernetes-enabled":
+			canonicalName = "enabled"
+		case "kubernetes-version":
+			canonicalName = "version"
+		}
+		if canonicalName == "" {
+			return nil, fmt.Errorf("field name %s not recognized", parts[0])
+		}
+		_, ok := values[canonicalName]
+		if !ok {
+			bufTemplate.WriteString(fmt.Sprintf(`%s "%s": {{ .%s }}`, commaOrEmptyString, canonicalName, canonicalName))
+			commaOrEmptyString = ","
+		}
+		values[canonicalName] = parts[1]
+	}
+
+	bufTemplate.WriteString("} }")
+	t := template.Must(template.New("template").Parse(bufTemplate.String()))
+	buf := new(bytes.Buffer)
+	err = t.Execute(buf, values)
+	if err != nil {
+		return nil, err
+	}
+	return buf, nil
+}
+
+func doApiCommand(cmd *cobra.Command, args []string) error {
+	var result []byte
+	var err error
+
+	if apiSettings.InputFile != "" {
+		if len(apiSettings.RawFields) > 0 || len(apiSettings.CookedFields) > 0 {
+			return fmt.Errorf("fields may not be specified when input from a file is specified")
+		}
+		if apiSettings.Method == "" {
+			apiSettings.Method = "PUT"
+		}
+		if len(apiSettings.RawFields) > 0 || len(apiSettings.CookedFields) > 0 {
+			return fmt.Errorf("fields may be specified only with an in-line body template, not input from a file")
+		}
+		contents, err := ioutil.ReadFile(apiSettings.InputFile)
+		if err != nil {
+			return err
+		}
+		result, err = doRequestWithPayload(apiSettings.Method, args[0], bytes.NewBuffer(contents))
```

Comment:
This panics if the user didn't provide any args.

Reply:
That's been fixed. There's even a test for it (`api: complains when no args are given`).

Reply:
This has been fixed since this review, with a test for it (https://github.com/rancher-sandbox/rancher-desktop/blob/66c3f4dbab1a500d4427f6254cdead4839e59953/e2e/rdctl.e2e.spec.ts#L251)